src/pikepdf/models/outlines.py

```diff
@@ -0,0 +1,247 @@
+from enum import Enum
+from itertools import chain
+
+from .. import Array, Dictionary, Name, Object
+
+
+class PageLocation(Enum):
+    XYZ = 1
+    Fit = 2
+    FitH = 3
+    FitV = 4
+    FitR = 5
+    FitB = 6
+    FitBH = 7
+    FitBV = 8
+
+
+PAGE_LOCATION_ARGS = {
+    PageLocation.XYZ: ('left', 'top', 'zoom'),
+    PageLocation.FitH: ('top',),
+    PageLocation.FitV: ('left',),
+    PageLocation.FitR: ('left', 'bottom', 'right', 'top'),
+    PageLocation.FitBH: ('top',),
+    PageLocation.FitBV: ('left',),
+}
+ALL_PAGE_LOCATION_KWARGS = set(chain.from_iterable(PAGE_LOCATION_ARGS.values()))
+
+
+def get_page_destination(pdf, page_num: int, page_location: (PageLocation, str) = None,
```

Comment:
I think `make_page_destination` would be a better name. I try to use `get_` when obtaining some data that already exists normally, and `make_` when we're constructing a new object.

Reply:
Done.